mesonbuild/interpreter/compiler.py

```diff
@@ -190,27 +190,27 @@ def __init__(self, compiler: 'Compiler', interpreter: 'Interpreter'):
     def compiler(self) -> 'Compiler':
         return self.held_object
 
-    def _dep_msg(self, deps: T.List['dependencies.Dependency'], endl: str) -> str:
+    def _dep_msg(self, deps: T.List['dependencies.Dependency'], compile_only: bool, endl: str) -> str:
         msg_single = 'with dependency {}'
         msg_many = 'with dependencies {}'
-        if not deps:
-            return endl
-        if endl is None:
-            endl = ''
         names = []
         for d in deps:
             if isinstance(d, dependencies.InternalDependency):
                 FeatureNew.single_use('compiler method "dependencies" kwarg with internal dep', '0.57.0', self.subproject,
                                       location=self.current_node)
                 continue
             if isinstance(d, dependencies.ExternalLibrary):
+                if compile_only:
+                    continue
```

Comment:
I'm wondering if we should even warn in this case, but that could be tricky because we don't want to warn in the case of `compile_only=self.compiler.is_cross`.

Reply:
I don't think we should do that, because `ExternalLibrary` can in the future also provide include flags: https://github.com/mesonbuild/meson/pull/10428 (and yes, that PR will need to change the compiler check message too).

I think it's one of those things that should at best be warned by a linter, because you don't always want to address it. For example, you have platform-specific deps that get added to a 'deps' array that is passed to all compiler checks. You don't want to have to maintain two separate lists.

Reply:
ok, let's ignore that then.